plugins/logs/plugin.go

```diff
@@ -425,6 +426,27 @@ func ParseConfig(config []byte, services []string, pluginList []string) (*Config
 		Parse()
 }
 
+func AllowConfigOverride(localConfig json.RawMessage) error {
+	if localConfig == nil {
+		return nil
+	}
+
+	type overridableConfig struct {
+		Reporting ReportingConfig `json:"reporting"`
+	}
+
+	d := json.NewDecoder(bytes.NewReader(localConfig))
+	d.DisallowUnknownFields()
+	var configStruct overridableConfig
+	err := d.Decode(&configStruct)
+
+	if err != nil {
+		return fmt.Errorf("discovery only allows manual configuration of the decision_logs plugin for the reporting.* configuration items")
```

Comment:
This is a bit odd. The code makes me think that there might be many ways to _fail_ `d.Decode()` and yet we pretend that it's always due to extra config items.

I suppose one alternative would be something along the lines of
```go
m := map[string]interface{}{}
if err := json.NewDecoder(...).Decode(&m); err != nil {
     return err
}
_, ok := m["reporting"]
if ok && len(m) == 1 {
    return nil
}
return fmt.Errorf("discovery only allows manual configuration of the decision_logs plugin for the reporting.* configuration items")
```

Reply:
yeah, this definitely needs some work. I originally wanted to only allow a subset of `reporting`, i.e. the buffer sizes and client rates. And agreed that there are other error scenarios, I was about to decode plainly and with disallowing unknown fields and then comparing if plain decoding works without error to narrow the root cause.

If we agree to make the entire runtime tree overridable, your suggestion is of course much simpler. I am not sure about the `Trigger` config inside `runtime`, wdyt?